VEX-in-Wolfi documentation

puerco opened the pull request:
This PR adds the first document detailing how VEX is originated and flows from wolfi config files all the way down to security scanners.

/cc @ltagliaferri @sheesh (for possible sync with academy)

Signed-off-by: Adolfo García Veytia (Puerco) <puerco@chainguard.dev>

lisa commented:
> This PR adds the first document detailing how VEX is originated and flows from wolfi config files all the way down to security scanners.
> 
> /cc @lisa @sheesh (for possible sync with academy)
> 
> Signed-off-by: Adolfo García Veytia (Puerco) [puerco@chainguard.dev](mailto:puerco@chainguard.dev)

# `@lisa` User Notification

Hello! You're receiving this notice because you've tagged me, @lisa, on GitHub, likely in error. Lisa is a common first name, and happens to be mine. I use the @lisa username here on GitHub. While being tagged in various repositories is an interesting way for me to discover new projects, it's likely not your intention. Did you mean to tag someone else? Did you mean to use `@lisa` instead, in a [word as word](https://en.wikipedia.org/wiki/Use%E2%80%93mention_distinction) usage?

# Next Steps

I'm unsubscribing from this issue, pull request or project so if you meant to tag someone else to notify them, I would recommend that. If you meant to use my user ID as a plain string, without having GitHub notify me, you can do so by wrapping with backticks (the \` character), like so: `@lisa`.

luhring commented on docs/vex.md:
This would be a good place for one or more links to more thorough explanations of VEX.

puerco replied:
I was planning to link to Academy, but I changed it to our blog post for now.

luhring commented on docs/vex.md:
nit:
```suggestion
By translating the Wolfi advisories into VEX metadata, the project can piece
```

luhring commented on docs/vex.md:
nit:
```suggestion
First, let’s look at what is inside the SBOM. The SBOM is attached to the image
```

luhring commented on docs/vex.md:
>While scanners implement native filtering

I think this phrase might be able to be misinterpreted as claiming that scanners implement native filtering _today_. I'd suggest something like:

>While scanners are preparing to support VEX filtering natively, ...

luhring reviewed (commented):
I really like this! 😍 

I have a few overall comments:

1. This explanation doc should probably go right **in the Wolfi repo**, not here in the `wolfictl` tool repo. It's describing a core philosophy and approach for Wolfi security.

2. Unless @kaniini has a different take, I'd suggest striking all "WolfiOS" appearances and always just saying "Wolfi". I think being consistent will make this more clear.

3. We should always capitalize "VEX": 1) to make it easier for readers to follow, and 2) to distinguish from the English (and unrelated) word "vex".

kaniini commented:
What @luhring said is right.  It's always "Wolfi", and never "Wolfi OS" or "WolfiOS".  The "os" repository is just one deliverable of the Wolfi project (and is the main one, of course).

puerco commented:
Cool, I was trying to somehow differentiate between wolfi the OS and wolfi the project. But I actually don't like WolfiOS at all.

I've addressed the nits, should I close this PR and reopen it somewhere in `os`? What do you think @kaniini and @luhring ?

luhring commented:
> I've addressed the nits, should I close this PR and reopen it somewhere in `os`? What do you think @kaniini and @luhring ?

Awesome! Yeah, that'd be my suggestion. "Where in `os`?" is a good question, since the root of the repo is a bit much. Maybe a `docs` folder for now? And at some point we could move the "How to Patch..." markdown file there, and others, too.

puerco commented:
Closing this PR as `wolfictl vex` was removed.
/cc @jspeed-meyers
